src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala

```diff
@@ -493,6 +493,143 @@ trait GoldTransforms extends SparkSessionWrapper {
+
+    val windowSpec1  = Window.partitionBy("commandId","clusterId","unixTimeMSStart","unixTimeMSEnd").orderBy(desc("uptime_in_state_H"))
+
+    val joinedDF_before_before = joinedDF.filter(state_before_before)
+      .withColumn("sum_total_DBU_cost",sum(col("total_DBU_cost")).over(windowSpec))
+      .withColumn("sum_uptime_in_state_H",sum(col("uptime_in_state_H")).over(windowSpec))
+      .withColumn("rn",row_number.over(windowSpec1))
+      .withColumn("rnk",rank.over(windowSpec1))
+      .filter('rnk === 1 && 'rn === 1).drop("rnk", "rn")
+
+
+    val joinedDF_after_before = joinedDF.filter(state_after_before)
+      .withColumn("sum_total_DBU_cost",sum(col("total_DBU_cost")).over(windowSpec))
+      .withColumn("sum_uptime_in_state_H",sum(col("uptime_in_state_H")).over(windowSpec))
+      .withColumn("rn",row_number.over(windowSpec))
+      .withColumn("rnk",rank.over(windowSpec))
+      .filter('rnk === 1 && 'rn === 1).drop("rnk", "rn")
+
+
+    val joinedDF_after = joinedDF.filter(state_after)
+      .withColumn("sum_total_DBU_cost",sum(col("total_DBU_cost")).over(windowSpec))
+      .withColumn("sum_uptime_in_state_H",sum(col("uptime_in_state_H")).over(windowSpec))
+      .withColumn("rn",row_number.over(windowSpec))
+      .withColumn("rnk",rank.over(windowSpec))
+      .filter('rnk === 1 && 'rn === 1).drop("rnk", "rn")
+
+
+    val joinedDF_before_after = joinedDF.filter(state_before_after)
+      .withColumn("sum_total_DBU_cost",sum(col("total_DBU_cost")).over(windowSpec))
+      .withColumn("sum_uptime_in_state_H",sum(col("uptime_in_state_H")).over(windowSpec))
+      .withColumn("rn",row_number.over(windowSpec))
+      .withColumn("rnk",rank.over(windowSpec))
+      .filter('rnk === 1 && 'rn === 1).drop("rnk", "rn")
+
+
+
+    val unionDF = joinedDF_before_before.union(joinedDF_after_before).union(joinedDF_after).union(joinedDF_before_after)
+      .withColumn("total_DBU_cost",sum(col("sum_total_DBU_cost")).over(windowSpec))
+      .withColumn("uptime_in_state_H",sum(col("sum_uptime_in_state_H")).over(windowSpec))
+      .withColumn("rn",row_number.over(windowSpec))
+      .withColumn("rnk",rank.over(windowSpec))
+      .drop("sum_total_DBU_cost","sum_uptime_in_state_H")
+
+    unionDF.cache()
+
+    val unionDF_final = unionDF.filter('rnk === 1 && 'rn === 1).drop("rnk", "rn")
+
+    val df = joinedDF.join(unionDF_final,Seq("unixTimeMSStart","unixTimeMSEnd","commandId"),"leftAnti")
+    val notebookClsfDF = unionDF_final.union(df.select(unionDF_final.schema.fieldNames.map(col): _*))
+
+    val notebookCodeAndMetaDF = notebookClsfDF.withColumnRenamed("unixTimeMS_state_start","unixTimeMS")
+      .toTSDF("unixTimeMS", "organization_id", "date","notebookId")
+            .lookupWhen(
+              notebookLookupTSDF,
+              maxLookAhead = 100
+            )
+      .df
+      .withColumn("total_dbu_cost_ps", col("total_DBU_cost") / col("uptime_in_state_H")/lit(3600))
```

Comment:
Semantically, this is not really total_dbu_cost_ps, this is more like average dbu cost per second, or just dbu_cost_ps, right?